src/js/BookReader/Mode1Up.js

```diff
@@ -178,13 +260,9 @@ export class Mode1Up {
    * @return {number}
    */
   getPageTop(index) {
-    const { floor } = Math;
-    const { book } = this;
-    let leafTop = 0;
-    for (const page of book.pagesIterator({ end: index, combineConsecutiveUnviewables: true })) {
-      leafTop += floor(page.height / this.br.reduce) + this.br.padding;
+    for (const {page, top} of this.pagesWithBounds()) {
+      if (page.index == index) return top;
```

Comment:
q: does we expect this to always find a match & return a value?

Reply:
Yep! Should always be given a valid index.